### PR TITLE
chore(all): Add GH Action to label PRs with template changes without tests

### PR DESCRIPTION
**Pull request by aroberts87**
## What this PR does / why we need it:
This PR introduces a new GitHub Action to analyze the contents of a PR and apply the `no-test` label if template files are modified without changes to test files.

## Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [x] Title of the PR starts with type and scope, (e.g. feat(agent,node-analyzer,sysdig-deploy):)
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers

Check Contribution guidelines in README.md for more insight.

### CODE REVIEW

**Review by ironashram — APPROVED** [review]
nice

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+jobs:
+  pr-label:
+    permissions:
+      contents: read
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/labeler@v4.0.2
+        with:
+          repo-token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (lilx1ao) [inline .github/workflows/labeler.yml:15]: generic token!

Comment (aroberts87) [reply .github/workflows/labeler.yml:15]: @lilx1ao - I'm not sure what your comment implies. Concern over using `secrets.GITHUB_TOKEN`? I'm following the documentation for using that GitHub supplied secret (https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret). You can even see their example on how to use it is actually showing how to use this labeler action too. 🙂

Comment (lilx1ao) [reply .github/workflows/labeler.yml:15]: ah ok 👍

**Review by lilx1ao — APPROVED** [review]

**Pull request merged**